checkov/terraform/graph_builder/foreach/builder.py

```diff
@@ -31,7 +31,5 @@ def handle(self, foreach_blocks: dict[str, list[int]]) -> None:
         if self._module_handler.local_graph.enable_modules_foreach_handling:
             if foreach_blocks.get(BlockType.MODULE):
                 self._module_handler.handle(foreach_blocks[BlockType.MODULE])
-                self._module_handler.local_graph._arrange_graph_data()
```

Comment:
Instead of duplicating all modules and then building edges we build edges on every iteration (see checkov/terraform/graph_builder/foreach/module_handler.py).